x-pack/test/api_integration/apis/management/rollup/rollup.js

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from 'expect.js';
+import { API_BASE_PATH, ROLLUP_INDEX_NAME } from './constants';
+
+import { registerHelpers } from './rollup.test_helpers';
+
+export default function ({ getService }) {
+  const supertest = getService('supertest');
+  const es = getService('es');
+
+  const {
+    createIndexWithMappings,
+    getJobPayload,
+    createJob,
+    deleteJob,
+    startJob,
+    stopJob,
+    cleanUp,
+  } = registerHelpers({ supertest, es });
+
+  describe('Rollup jobs', () => {
+    after(() => cleanUp());
+
+    describe('Rollup indices', () => {
+      it('should return an empty object when there are no rollup indices', async () => {
+        const uri = `${API_BASE_PATH}/indices`;
+
+        const { body } = await supertest
+          .get(uri)
+          .expect(200);
+
+        expect(body).to.eql({});
+      });
+    });
+
+    describe('Index patterns', () => {
+      it('should return the date, numeric and keyword fields when an index (pattern) is found', async () => {
+        const indexName = await createIndexWithMappings();
+
+        const uri = `${API_BASE_PATH}/index_pattern_validity/${indexName}`;
+
+        const { body } = await supertest
+          .get(uri)
+          .expect(200);
+
+        expect(body).to.eql({
+          dateFields: ['testCreatedField'],
+          keywordFields: ['testTagField'],
+          numericFields: ['testTotalField'],
+          doesMatchIndices: true,
+          doesMatchRollupIndices: false,
+        });
+      });
+
+      it('should not return any fields when no index (pattern) is found', async () => {
+        const uri = `${API_BASE_PATH}/index_pattern_validity/index-does-not-exist`;
+
+        const { body } = await supertest
+          .get(uri)
+          .expect(200);
+
+        expect(body).to.eql({
+          dateFields: [],
+          keywordFields: [],
+          numericFields: [],
+          doesMatchIndices: false,
+          doesMatchRollupIndices: false,
+        });
+      });
+    });
+
+    describe('Crud', () => {
+      describe('list', () => {
+        it('should return an empty array when there are no jobs', async () => {
+          const { body } = await supertest
+            .get(`${API_BASE_PATH}/jobs`)
+            .expect(200);
+
+          expect(body).to.eql({ jobs: [] });
+        });
+      });
+
+      describe('create', () => {
+        it('should create a rollup job', async () => {
+          const indexName = await createIndexWithMappings();
```

Comment:
Do we need to create new indices for these tests? Can we create a single index in a `beforeEach` and target that index in all of the tests? This would speed things up slightly and also simplify the tests.